### PR TITLE
Enable all test methods in ConcurrentLoginTest for JPA Map Storage

### LINKED ISSUE #12707: Enable ConcurrentLoginTest for JPA map storages
`ConcurrentLoginTest.concurrentLoginSingleUser` and `ConcurrentLoginTest.concurrentLoginMultipleUsers` were disabled for CHM because they were failing. The tests should be re-enabled for HotRod and JPA map storages. Additionally we could investigate exact cause why they are failing for CHM and try to fix that.

### LINKED ISSUE #13210: JPA Map Storage with CRDB: ConcurrentLoginTest failures
### Describe the bug

**org.keycloak.testsuite.admin.concurrency.ConcurrentLoginTest**  presents failures/errors with `CockroachDB`:

```
[ERROR]   ConcurrentLoginTest.concurrentLoginSingleUserSingleClient:170->AbstractConcurrencyTest.run:63->AbstractConcurrencyTest.run:122 » Runtime There were failures in threads. Failures count: 2
```

- Required JPA stores to reproduce the failure: **authenticationSessions**

```
[ERROR]   ConcurrentLoginTest.concurrentCodeReuseShouldFail:249->AbstractConcurrencyTest.run:63->AbstractConcurrencyTest.run:122 » Runtime There were failures in threads. Failures count: 2
```

- Required JPA stores to reproduce the failure: **userSessions**

For both tests, logs show the following error:

```
Caused by: org.postgresql.util.PSQLException: ERROR: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_WRITE_TOO_OLD - WriteTooOld flag converted to WriteTooOldError): "sql txn" meta={id=7af23746 key=/Table/3142/1/"\x9cX4\x9e\xaf\x94F/\xac\xe41~H@\"\x10"/0 pri=0.05982192 epo=0 ts=1658239825.904432104,1 min=1658239825.904379733,0 seq=2} lock=true stat=PENDING rts=1658239825.904379733,0 wto=false gul=1658239826.404379733,0
  Hint: See: https://www.cockroachlabs.com/docs/v21.2/transaction-retry-error-reference.html#retry_write_too_old
```

### Version

main

### Expected behavior

Tests should run without failures or errors

### Actual behavior

_No response_

### How to Reproduce?

1- Setup CockroachDB according to the instructions in https://github.com/keycloak/keycloak/discussions/9596
2- Run the test with the correct parameters, enabling JPA storage for **authenticationSessions**/**userSessions** only:

```
cd testsuite/integration-arquillian

mvn clean install -Pmap-storage -Dpageload.timeout=90000 -Dkeycloak.authSession.map.storage.provider=jpa \
-Dkeycloak.userSession.map.storage.provider=jpa \
-Dkeycloak.map.storage.connectionsJpa.url=jdbc:postgresql://localhost:26257/keycloak \
-Dkeycloak.map.storage.connectionsJpa.user=keycloak -Dkeycloak.map.storage.connectionsJpa.password=pass \
-Dtest=ConcurrentLoginTest
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sguilhen**
- Tests still disabled for Hotrod and CHM
- Fixes concurrent login issues with CRDB. Verified with both PostgreSQL and CockroachDB.

Closes #12707
Closes #13210

### CODE REVIEW

**Review by ahus1 — CHANGES_REQUESTED** [review]
Hi @sguilhen - looks good to me, one some change suggested, see below.

File: services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java [hunk 1]
```diff
@@ -822,6 +823,9 @@ public Response handleBrowserException(Exception failure) {
                 return ErrorPage.error(session, authenticationSession, Response.Status.BAD_REQUEST, Messages.INVALID_USER);
             }
 
+        } else if (KeycloakModelUtils.isExceptionRetriable(failure)) {
+            // let calling code decide if whole action should be retried.
+            throw (RuntimeException) failure;
```
[the remark this answers is not in the document]
Comment (sguilhen) [reply services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java:828]: > IMHO one can't be sure that `failure` is a `RuntimeException`, although the current implementation suggests this.

Well, currently a retriable exception must be a SQLException whose state is 40001, and that is a RuntimeException. So if `isExceptionRetriable()` returns true we know the exception is a runtime one.

[the remark this answers is not in the document]
Comment (sguilhen) [reply services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java:828]: EDIT: actually it is not a RuntimeException, it is often encapsulated as one though. I'll change the code to follow your suggestion.

[the remark this answers is not in the document]
Comment (ahus1) [reply services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java:828]: Hm, it is actually testing that the root exception is a `SQLException`. The wrapping exception can be anything, although its wrappers will most likely be a runtime exception as well.

[the remark this answers is not in the document]
Comment (ahus1) [reply services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java:828]: Ah, now I see the edit; agreed. Please disregard my previous comment.

**Review by ahus1 — APPROVED** [review]
Thank you for this PR - this is a huge step forward on the road to CockroachDB support!

**Review by vramik — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]
Approving per @vramik 's and @ahus1 's reviews

**Pull request merged**